src/fileioc/fileioc.asm

```diff
@@ -1621,19 +1643,28 @@ util_set_offset:
 	ld	(hl), bc
 	ret
 
-util_Arc_Unarc:				; properly handle garbage collects
+util_archive:				; properly handle garbage collects
+	ld	iy, ti.flags
+	call	ti.ChkFindSym
 	call	ti.ChkInRam
-	jp	nz,ti.Arc_Unarc		; if the file is already in archive, we won't trigger a gc
-	ex	hl,de
-	call	ti.LoadDEInd_s
-	ex	hl,de
-	call	util_ArchiveHasRoom
-	jp	nz,ti.Arc_Unarc		; gc will not be triggered
+	ret	nz
+	call	ti_ArchiveHasRoomVar.entry_sym
+	jp	nz,ti.Arc_Unarc
+	call	ti.PushOP1
 	call	util_pre_gc_default_handler
 util_pre_gc_handler := $-3
 	call	ti.Arc_Unarc
```

Comment:
Should there be a pop/push OP1 before Arc_Unarc, in case the pre-GC handler trashes it for some reason?